.github/workflows/default.yaml

```
@@ -109,10 +109,35 @@ jobs:
           sudo apt-get --quiet=2 build-dep squid
           sudo apt-get --quiet=2 install linuxdoc-tools libtool-bin
 
+      - name: Install prerequisite MacOS packages
+        if: runner.os == 'macOS'
+        run: |
+          brew install \
+            automake coreutils cppunit gawk \
+            gnu-getopt gnu-sed grep libtool \
+            make openldap openssl@3 cyrus-sasl
+
       - name: Checkout sources
         uses: actions/checkout@v4
 
-      - run: ./test-builds.sh
+      - name: Run build on Linux
+        if: runner.os == 'Linux'
+        run: ./test-builds.sh
+
+      - name: Run build on MacOS
+        if: runner.os == 'macOS'
+        run: |
+          eval `brew shellenv`
+          export PKG_CONFIG_PATH="$HOMEBREW_PREFIX/lib/pkgconfig:$HOMEBREW_PREFIX/opt/openldap/lib/pkgconfig:$HOMEBREW_PREFIX/opt/cyrus-sasl/lib/pkgconfig"
+          export GETOPT="$HOMEBREW_PREFIX/opt/gnu-getopt/bin/getopt"
+          if gcc --version | grep -q clang ; then
+            export CFLAGS="-Wno-compound-token-split-by-macro${CFLAGS+ $CFLAGS}" # needed fir ltdl
+          fi
+          export pjobs="-j`nproc`"
+          export MAKE="$HOMEBREW_PREFIX/bin/gmake"
+          export CPPFLAGS="-I$HOMEBREW_PREFIX/include${CPPFLAGS+ $CPPFLAGS}" # openssl fails without this
```

Is `-I$HOMEBREW_PREFIX/include` missing from the OpenSSL pkg-config configuration provided by homebrew or is it there, but we are mishandling it somehow? Just curious...

---

Ouch. I am surprised such a bug has not been fixed upstream yet.

If OpenSSL v1 pkg-config configuration has similar problems, then I suggest:

```suggestion
           # This is done to keep in sync with LDFLAGS below
          export CPPFLAGS="-I$HOMEBREW_PREFIX/include${CPPFLAGS+ $CPPFLAGS}"
           # `pkg-config --libs openssl` points to a wrong directory (missing lib/)
```

Reordering CPPFLAGS and LDFLAGS blobs (and s/below/above/) is probably a good idea in that case.

Otherwise, if we can stick with OpenSSL v1 (if it is available by default and "works" without complications), then we should probably test with that version for now.